src/fswAlgorithms/effectorInterfaces/solarArrayReference/solarArrayReference.rst

```diff
@@ -0,0 +1,90 @@
+
+
+Detailed Module Description
+---------------------------
+For this module to operate, the user needs to provide two unit directions as inputs:
+
+- :math:`{}^\mathcal{B}\boldsymbol{\hat{a}}_1`: direction of the solar array drive, about which the rotation happens;
+- :math:`{}^\mathcal{B}\boldsymbol{\hat{a}}_2`: direction perpendicular to the solar array surface, with the array at a zero rotation.
+
+To compute the reference rotation :math:`\theta_R`, the module computes the unit vector :math:`{}^\mathcal{R}\boldsymbol{\hat{a}}_2`, which is coplanar with 
+:math:`{}^\mathcal{B}\boldsymbol{\hat{a}}_1` and the Sun direction :math:`{}^\mathcal{R}\boldsymbol{\hat{r}}_S`. This is obtained as:
+
+.. math::
+    {}^\mathcal{R}\boldsymbol{a}_2 = {}^\mathcal{R}\boldsymbol{\hat{r}}_S - ({}^\mathcal{R}\boldsymbol{\hat{r}}_S \cdot {}^\mathcal{B}\boldsymbol{\hat{a}}_1) {}^\mathcal{B}\boldsymbol{\hat{a}}_1
+
+and then normalizing to obtain :math:`{}^\mathcal{R}\boldsymbol{\hat{a}}_2`. The reference angle :math:`\theta_R` is the angle between :math:`{}^\mathcal{B}\boldsymbol{\hat{a}}_2` and :math:`{}^\mathcal{R}\boldsymbol{\hat{a}}_2`:
+
+.. math::
+    \theta_R = \arccos ({}^\mathcal{B}\boldsymbol{\hat{a}}_2 \cdot {}^\mathcal{R}\boldsymbol{\hat{a}}_2).
+
+The same math applies to the case where the body reference is used. In that case, the same vectors are expressed in body-frame coordinates. Note that the unit directions :math:`\boldsymbol{\hat{a}}_i` have the same components in both the body and reference frame, because they are body-fixed and rotate with the spacecraft hub.
+
+Some logic is implemented such that the computed reference angle :math:`\theta_R` and the current rotation angle :math:`\theta_C` received as input from the ``hingedRigidBodyInMsg`` are never more than 360 degrees apart.
+
+The derivative of the reference angle :math:`\dot{\theta}_R` is computed via finite differences.
+
+
+User Guide
+----------
+The required module configuration is::
+
+    solarArrayConfig = solarArrayRotation.solarArrayRotationConfig()
+    solarArrayWrap = unitTestSim.setModelDataWrap(solarArrayConfig)
+    solarArrayWrap.ModelTag = "solarArrayRotation"  
+    solarArrayConfig.a1_B = [1, 0, 0]
+    solarArrayConfig.a2_B = [0, 0, 1]
+    solarArrayConfig.bodyFrame = 0
+    unitTestSim.AddModelToTask(unitTaskName, solarArrayWrap, solarArrayConfig)
+	
+The module is configurable with the following parameters:
+
+.. list-table:: Module Parameters
+   :widths: 34 66
+   :header-rows: 1
+
+   * - Parameter
+     - Description
+   * - ``a1_B``
```

Comment:
These unit vector variables should be called `a1Hat_B` and `a2Hat_B`

Reply:
Fixed it.